sunpy/visualization/animator/base.py

```diff
@@ -522,55 +500,112 @@ def _sanitize_axis_ranges(self, axis_ranges, data):
             * array of axis length, check that it was passed for a slider axes and do nothing
             if it was, error if it is not.
         """
+        ndim = len(data_shape)
         # If no axis range at all make it all [min,max] pairs
         if axis_ranges is None:
-            axis_ranges = [None] * data.ndim
+            axis_ranges = [None] * ndim
 
         # need the same number of axis ranges as axes
-        if len(axis_ranges) != data.ndim:
+        if len(axis_ranges) != ndim:
             raise ValueError("Length of axis_ranges must equal number of axes")
 
-        # For each axis validate and translate the axis_ranges
-        for i, d in enumerate(data.shape):
-            # If [min,max] pair or None
-            if axis_ranges[i] is None or (len(axis_ranges[i]) == 2 and d > 2):
-                # If min==max or None
-                if axis_ranges[i] is None or axis_ranges[i][0] == axis_ranges[i][1]:
-                    if i in self.slider_axes:
-                        axis_ranges[i] = np.arange(0, d+1)
-                    else:
-                        axis_ranges[i] = [0, d]
-                        # min max pair for slider axes should be converted
-                        # to an array
-                elif i in self.slider_axes:
-                    axis_ranges[i] = np.arange(axis_ranges[i][0], axis_ranges[i][1]+1)
-
-            # If we have a whole list of values for the axis, make sure we are a slider axis.
-            elif len(axis_ranges[i]) == d or axis_ranges[i].shape == data.shape:
-                axis_ranges[i] = np.array(axis_ranges[i])
-
-            # If above criteria are not met, raise an error.
+        # Define error message for incompatible axis_range input.
+        def incompatible_axis_ranges_error_message(j):
```

Comment:
Any reason this was defined inside `_sanitize_axis_ranges`?

Reply:
Oh.  I think this was originally a lambda function as all it does is create a string.  But previous comments said we don't like lambda functions for some reason...

Reply:
No, I just wanted to know why. If it needs to be defined, there, we can keep it.

Reply:
In fact I think it's better to go back to a lambda function as it's not used anywhere else except inside ```_sanitize_axis_ranges``` and only defines a string.